### PR TITLE
ARROW-16347: [Release] Escape backtick in verification script

### PR DESCRIPTION
**Pull request by kszucs**

### CODE REVIEW

**Comment by raulcd** [comment]
I think there was an open ticket for this:
https://issues.apache.org/jira/browse/ARROW-16347
The fix looks good to me, validated locally :+1:

**Review by kszucs — COMMENTED** [review]
+1

File: dev/release/verify-release-candidate.sh [hunk 1]
```diff
@@ -469,7 +469,7 @@ maybe_setup_conda() {
     conda activate $env
   elif [ ! -z ${CONDA_PREFIX} ]; then
     echo "Conda environment is active despite that USE_CONDA is set to 0."
-    echo "Deactivate the environment using `conda deactive` before running the verification script."
+    echo "Deactivate the environment using \`conda deactive\` before running the verification script."
```
Comment (raulcd) [inline dev/release/verify-release-candidate.sh:472]:
```suggestion
    echo "Deactivate the environment using \`conda deactivate\` before running the verification script."
```

**Review by raulcd — CHANGES_REQUESTED** [review]

File: dev/release/verify-release-candidate.sh (hunk 1 above)
Comment (kszucs) [reply dev/release/verify-release-candidate.sh]: Thanks for catching that!

**Pull request closed without merging**